code-challenges.test.js

```diff
@@ -24,38 +33,225 @@ var secretCodeWord2 = "Gobbledygook"
+    //iterate with for loop
+    //use if statement - if [i] = ('a') return 4 etc.
+    //return array to string and return string
+//New process because that didn't work and I got lost in an overly complex thought process trying to force it to work and realized there must be a simpler way -
+    //google
+        //use .replace(): string.replace(searchvalue, newvalue)
+            //**make a notecard: 
+                    //The replace() method searches a string for a specified value, or a regular expression, and returns a new string where the specified values are replaced. */
+                //Make another notecard for things that become more relevant when you actually read them: 
+                    //Note: If you are replacing a value (and not a regular expression), only the first instance of the value will be replaced. To replace all occurrences of a specified value, use the global (g) modifier (see "More Examples" below).
 
 // b) Create the function that makes the test pass.
 
+// const encoder = (string) => {
+//     let stringA = string.replace(/[aA]/g, "4")
+//     let stringE = stringA.replace(/[eE]/g, "3")
+//     let stringI = stringE.replace(/[iI]/g, "1")
+//     let stringO = stringI.replace(/[oO]/g, "0")
+//     return stringO
+// }
+// console.log(encoder(secretCodeWord1))
+// Test Suites: 1 passed, 1 total
+// Tests:       1 passed, 1 total
+    //this works but doesn't seem very dynamic
+        //feels like manually forcing it to go through a for loop and splitting into an array and mapping over should have worked and resulted in cleaner code
 
+// const encoder = (string) => {
+//     let array = string.split("")
+//     return array
+// }
+// console.log(encoder(secretCodeWord1))
+// [
+//     'L', 'a', 'c', 'k',
+//     'a', 'd', 'a', 'i',
+//     's', 'i', 'c', 'a',
+//     'l'
+//   ]
+// const encoder = (string) => {
+//     let array = string.split("")
+//     for(let i = 0; i<array.length; i++){
+//         array[i] = array[i].replace('a', '4')
+//     }
+//     return array
+// }
+// console.log(encoder(secretCodeWord1))
+// [
+//     'L', '4', 'c', 'k',
+//     '4', 'd', '4', 'i',
+//     's', 'i', 'c', '4',
+//     'l'
+//   ]
+// const encoder = (string) => {
+//     let array = string.split("")
+//     for(let i = 0; i<array.length; i++){
+//         array[i] = array[i].replace('a', '4'; 'i', '1')
+//     }
+//     return array
+// }
+// console.log(encoder(secretCodeWord1))
+//close but the syntax isnt right
+// const encoder = (string) => {
+//     let array = string.split("")
+//     for(let i = 0; i<array.length; i++){
+//         array[i] = array[i].replace(/'a':'4', 'i':'1'/)
+//     }
+//     return array
+// }
+// console.log(encoder(secretCodeWord1))
+//noop
+const encoder = (string) => {
+    let array = string.split("")
+    for(let i = 0; i<array.length; i++){
+        array[i] = array[i].replace('a', '4')
+        array[i] = array[i].replace('i', '1')
+    }
+    return array.join("")
+}
```

Comment:
alright! this is good --  but it's only half the problem. 
Your logic makes sense so I'll just add in here you need. 

```javascript
array[i] = array[i].replace('o, '0')
array[i] = array[i].replace('e, '3')
```
and maybe a .toLowerCase() somewhere in there